libraries/adaptive-expressions/src/builtinFunctions/lastIndexOf.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @module adaptive-expressions
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ExpressionEvaluator } from '../expressionEvaluator';
+import { Expression } from '../expression';
+import { ReturnType } from '../returnType';
+import { ExpressionType } from '../expressionType';
+import { FunctionUtils } from '../functionUtils';
+import { MemoryInterface } from '../memory/memoryInterface';
+import { Options } from '../options';
+
+/**
+ * Return the starting position or index value of the last occurrence of a substring.
```

Comment:
>starting position or index value of the last occurrence of a substring [](start = 14, length = 70)

index of the last value occurrence found in string.

Reply:
OK, this is still not correct.  
For strings returns the last index of the last occurrence of a substring.
For arrays return the last index of a value in the array.
Indices start at 0 and -1 is returned if not found.

---
In reply to: [454715923](https://github.com/microsoft/botbuilder-js/pull/2502#discussion_r454715923) [](ancestors = 454715923)